src/comm.F

```diff
@@ -13817,7 +13822,10 @@ subroutine prepcorners_GPU(s,nw1,nw2,ne1,ne2,sw1,sw2,se1,se2,  &
 
       integer :: i,j
       logical, parameter :: Debug = .FALSE.
-      !$acc declare present(s)
+
+      !$acc data present(s,nw1,nw2,ne1,ne2,sw1,sw2,se1,se2, &
+      !$acc              pw1,pw2,pe1,pe2,ps1,ps2,pn1,pn2)
+
```

Comment:
@sjsprecious, I don't understand the purpose of this set of directives for only a small region. If we want to make sure that data is present why are we not using a declare present() clause like is used throughout the rest of CM1.

Reply:
Thanks @johnmauff. I think it is equivalent to use `acc declare present (s,nw1,...)` here but it is just my personal preference to use `acc data present` directive. I used to use `acc declare` for module variables only, which helps me avoid some confusions.

Reply:
I guess I have the opposite personal preference.  In particular, it requires an $acc end data which seems strange because the variable does not stop being present right.  It is a persistent variable on the device.  @sjsprecious Is there an issue with OpenMP offload syntax that would suggest we should use one versus the other?

Reply:
OpenMP offload syntax does not have a `present` clause right now. It uses `data map (to|from|alloc)` directive for data transfer, which is translated to `acc data copyin|copyout|create` directive in OpenACC. 
Regarding the `omp declare target` directive, there is no clause like `omp declare target create` in OpenMP compared to the `acc declare create` in OpenACC. Thus there is always data transfer when we use `omp declare target` directive, which could cause undesired effect.

Reply:
In addition, `$acc end data` here does not mean that the variables will not be present after this subroutine.
Adding `acc data` here is just a sanity check to me since sometimes `default(present)` clause is missing from the `acc parallel` directive and we may accidentally cause some unexpected data movement.

Reply:
Ok, thanks for the clarification.